lib/stripe/payment_intent_validator.rb

```diff
@@ -3,9 +3,11 @@
 # This class validates if a given payment intent ID is valid in Stripe
 module Stripe
   class PaymentIntentValidator
-    def call(payment_intent_id, stripe_account_id)
-      payment_intent_response = Stripe::PaymentIntent.retrieve(payment_intent_id,
-                                                               stripe_account: stripe_account_id)
+    def call(payment)
```

Comment:
stripe_account_id could have been left as optional param and used from the stripe_sca class to avoid some extra db queries.

Reply:
It removes one query from `StripeSCA` at the same time as adding one query in the service, so it's net zero. In theory we could add some memoizing in `StripeSCA`, but I wasn't sure it was a good idea.